.github/workflows/ci-js.yml

```diff
@@ -0,0 +1,78 @@
+name: CI
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+on:
+  push:
+    branches: ["main"]
+  pull_request:
+    types: [opened, edited, synchronize]
+
+jobs:
+  build:
+    name: build and test
+    timeout-minutes: 15
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [
+          ubuntu-latest,
+          macos-latest,
+          # windows-latest,
```

Comment:
I'd like to come back to this

Reply:
A comment detailing what issues were encountered would be helpful

Reply:
It didn't seem to pass in CI and I wanted to isolate the issue and fix. I'm not even sure at this point it's a real problem